code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -743,3 +743,26 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 		current_reaction.react(air_mixture = src, working_power = working_power, electrolyzer_args = electrolyzer_args)
 
 	garbage_collect()
+
+/// Convert a gas mixture to a string (ie. "o2=22;n2=82;TEMP=180")
+/// Rounds all temperature and gases to 0.01 and skips any gases less than that amount
+/datum/gas_mixture/proc/to_string()
+	var/list/cached_gases = gases
+	var/rounded_temp = round(temperature, 0.01)
+
+	var/list/atmos_contents = list()
+	var/temperature_str = "TEMP=[num2text(rounded_temp)]"
```

Comment:
doesn't temp need a ; attached at the end of it?

Reply:
Nothing I could find showed me that it needed a ; at the end.

https://github.com/tgstation/tgstation/blob/167a682f89d833cddbb8e65a38328c00b591a93a/code/__DEFINES/atmospherics/atmos_mapping_helpers.dm#L3-L52

I did refactor https://github.com/tgstation/tgstation/pull/91051/commits/7d39462b2bb27822764bad5017cc044cfdee69e8 so that we used the `list.Join(";")` method.

Reply:
oh it goes on the end, excuse me